src/libclipper/include/clipper/threadpool.hpp

```diff
@@ -0,0 +1,269 @@
+#ifndef CLIPPER_LIB_THREADPOOL_HPP
```

Comment:
Added a reference link in the comments. Only things changed were some variable names and I added unit tests.